Friendly error message for non-existent file/directory

anshulvermapatel opened the pull request:
Fixes https://github.com/helm/helm/issues/6713

Have updated the error message for the missing Chart.yaml

PS - This is my first PR here. Do let me know if I need to make any further changes.
cc @karuppiah7890

anshulvermapatel committed:
Friendly error message for non-existent file/directory

Signed-off-by: Anshul Verma <anshulvermapatel@gmail.com>

Friendly error message for non-existent file/directory

Signed-off-by: Anshul Verma <anshulvermapatel@gmail.com>

Friendly error message for non-existent file/directory

Signed-off-by: Anshul Verma <anshulvermapatel@gmail.com>

bacongobbler commented on pkg/chartutil/chartfile.go:
```diff
@@ -74,7 +74,7 @@ func IsChartDir(dirName string) (bool, error) {
 	}
 	if chartContent == nil {
-		return false, errors.Errorf("chart metadata (%s) missing", ChartfileName)
+		return false, errors.Errorf("(%s) : No such file or directory", ChartfileName)
```
This doesn't seem right. A Chart.yaml file that has no content (e.g. `echo "" > mychart/Chart.yaml`) can trigger this error, which means the error is misleading.

Can you please write a unit test in `chartfile_test.go` that covers the case you are trying to fix?

sayboras committed:
Used timeout instead of deadline

Signed-off-by: sayboras <sayboras@yahoo.com>

hangpark committed:
fix(pkg/downloader): add failing test for build with repo alias

Signed-off-by: Hang Park <hangpark@kaist.ac.kr>

hangpark committed:
fix(pkg/downloader): resolve repo alias before checking digests on build

`Update()` gets repo names before resolving a lock file by calling
`resolveRepoNames(req)`. But that method changes aliased repo URLs into
the actual URLs. That makes digests from `helm update` and `helm build`
be different for each other.

To make them in sync, setting actual (resolved) repo URLs into the
loaded chart during `helm build` is necessary. Thus, this commit adds an
extra step in the `Build()` implementation.

For comments, this commit also changes the name of `getRepoNames()` into
`resolveRepoNames()` to avoid misunderstanding since getters are
expected to not mutate their input data in general.

Signed-off-by: Hang Park <hangpark@kaist.ac.kr>

Commit:
fix(get): install Helm v2.16.1

This is a temporary fix. This prevents the get script from installing Helm 3 as soon as it's released, potentially causing disruption to users that were expecting a Helm 2 release.

A `get-helm-3` script is also supplied, which is identical to the previous `get` script. That way, Helm 3 users also have an equivalent way to install Helm 3.

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

thomastaylor312 committed:
fix(wait): Adds support for waiting on v1 apiextensions for CRDs

This was a missed update when we updated the k8s libraries. I validated
that this works for CRD installs with v1beta1 and v1

Signed-off-by: Taylor Thomas <taylor.thomas@microsoft.com>

Commit:
fix(get): hard code DESIRED_VERSION when unset

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

Commit:
fix(get-helm-3): remove tiller checks, fixup version check

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

anshulvermapatel commented:
Hi Matthew,
I am sorry, I messed it up on this one, I am closing this one to create a fresh clean one.
Sorry again.
